### PR TITLE
incorporate changes of MVPA Light toolbox

### PR DESCRIPTION
**Pull request by treder**
Hi Jan-Mathijs, this is my first draft at the update of ft_statistics_mvpa.m , maybe we can iterate from here?

The MVPA-Light part is now very short and at the same time more powerful (should in principle work for arbitrary number of dimensions, and any of the data dimensions can be used as features or for generalization). The lines I've reworked/added:
- ll.1 - 192
- ll.245 - 280

The parts I didn't touch (where anyways you contributed most of the code):
- dealing with connectivity 
- reshaping the 'perf' output of MVPA-Light back into a form that FT likes
- adj_dim 
- cfg.mvpa.timwin: this is not something MVPA-Light uses, is this required? (l.224+)
- dealing with the .neighbours field

I will have to work through the FT tutorial again once this is solved. The cross-validation and time classification should still work (the mv_classify call works fine, but it breaks further down since I don't know what to do with adj_dim). But the searchlight and time generalization have a different syntax now in order to be more flexible / general.

### CODE REVIEW

**Comment by treder** [comment]
I've just realised there are other differences between my version and the FT Master one, like the use of the shiftdim function at the bottom of the code. Is my version outdated? (I did run git fetch upstream on my fork)

**Comment by schoffelen** [comment]
Looking at the diff I am a bit worried indeed that you have been working on an outdated version, which makes it hard to get a feeling of what's going on with your changes w.r.t. the version that I have working for myself. On  the other hand it seems that so far the major change in your version is that you commented out a large chunk of code, right? :) What I find important for my own spielerei is that I'd still be able to do the efficient naive_bayes classification with a spatiotemporal searchlight (as implemented with the cfg.mvpa.neighbours as a cell-array. 

We could also take it from here indeed, but either way I'd be in favour of you trying to PR from a topic branch which is not called master (so that it is easier for me to pull into my own repo without the need to think about a local branch which has a different name. Would it be a big deal for you to move the changed function into another branch, and file a new PR?

Next thing to do would be (I think) to follow the tutorial, and titrate the code such, that the options highlighted will work (again). I don't mind contributing to that

**Comment by schoffelen** [comment]
I am quite sure that I worked on the tutorial before, and might have an updated version somewhere on a topic branch. I would like to push this first to the website, so as to not lose that work. I'll keep you posted on this.

**Comment by schoffelen** [comment]
Hmm, I cannot find back what I did a while back. That was a waste of work then apparently. Story of my life...

**Comment by treder** [comment]
Good plan. I'll move everything to a feature branch and make sure it's matched with the newest version. There'll be some extra code to ensure downward compatibility with the current version of the tutorial, although I suppose we'd later update the tutorial to be in line with the changed interface. 

Regarding your spielerei with the spatiotemporal searchlight do you happen to have a code snippet I can use with the FT example data to test this?

**Comment by schoffelen** [comment]
> Good plan. I'll move everything to a feature branch and make sure it's matched with the newest version. There'll be some extra code to ensure downward compatibility with the current version of the tutorial, although I suppose we'd later update the tutorial to be in line with the changed interface.
> 
> Regarding your spielerei with the spatiotemporal searchlight do you happen to have a code snippet I can use with the FT example data to test this?

aber sicher! I used to have it in an updated version of the mvpa tutorial, but this got lost. give me some time to whip something up.

**Comment by schoffelen** [comment]
% script to demonstrate to Matthias, how I would like to be able to do a
% spatio-temporal searchlight analysis on (e.g. MEG sensor data


% specify some sane defaults for the mvpa. NOTE: for an lda probably one
% would want to specify regularization parameter etc. perhaps.
mvpa            = [];
mvpa.classifier = ft_getopt(mvpa, 'classifier', 'naive_bayes');
mvpa.metric     = ft_getopt(mvpa, 'metric',     'accuracy');
mvpa.repeat     = ft_getopt(mvpa, 'repeat',     5);
mvpa.timwin     = ft_getopt(mvpa, 'timwin',     30);

% ensure that MVPA-light is on the path
addpath('~/matlab/toolboxes/MVPA-light/startup/');
startup_MVPA_Light;
% FIXME the above is not dealt with in ft_hastoolbox, i.e. ft_hastoolbox
% does not work well, even with the folder on the path

% take the data from the eventrelatedaveraging tutorial
datadir = '/home/common/matlab/fieldtrip/data/ftp/tutorial/eventrelatedaveraging';
load(fullfile(datadir,'dataFC_LP'));
load(fullfile(datadir,'dataFIC_LP'));
load(fullfile(datadir,'dataIC_LP'));

data   = ft_appenddata([], dataFC_LP, dataFIC_LP);
design = [ones(1,numel(dataFC_LP.trial)) ones(1,numel(dataFIC_LP.trial))*2];

% create a spatial neighbourhood structure for the spatial searchlight
pw_dir = pwd;
[ft_ver, ft_path] = ft_version;
cd(fullfile(ft_path, 'private'));
load ctf151_neighb;
cfg = [];
cfg.neighbours = neighbours;

% create planar gradient data, to make neighbours more meaningful
cfg.planarmethod = 'sincos';
data = ft_megplanar(cfg, data);

cfg = [];
cfg.neighbours = neighbours;
if numel(data.label)>151
  % data is probably planar gradient
  isplanar = true;
end

if isplanar
  n = numel(data.label)./2; % data is planar gradient
  cfg.channel = data.label(1:n);
  cfg.channel = strrep(cfg.channel, '_dH', '');
  label = cfg.channel;
else
  cfg.channel = data.label;
end

C = channelconnectivity(cfg);
if isplanar 
  C = [C C]; % define _dH and _dV as neighbours, and go from 298 back to 149 of the original labels
end
cd(pw_dir);

% decrease the memory footprint of the data
cfg         = [];
cfg.latency = [-0.1 1];
data        = ft_selectdata(cfg, data);
% % specify the prior for the naive_bayes in case of unbalanced designs
% if isequal(mvpa.classifier, 'naive_bayes')
%   mvpa.hyperparameter.prior = Nx; % log will be done in the function
% end
param = [];
param.oversample_test_set = true;

% use oversampling on both training and test set, due to disbalance across
% conditions
mvpa.preprocess = {'oversample'};
mvpa.preprocess_param{1} = param;


cfg              = [];
cfg.method       = 'mvpa';
cfg.design       = design;
cfg.mvpa         = mvpa;
cfg.connectivity = C; % assumes channels in order _dH, _dV per chunk
stat             = ft_timelockstatistics(cfg, data);
stat.label       = dataFC_LP.label;% still there's something going wrong with the assignment of the label in the output


cfg.connectivity = eye(size(C,1)*2); % there's something amiss with the cfg handling at the moment, so in order to demonstrate time-only, or space-only searchlight, the connectivity still needs to be defined
stat_time = ft_timelockstatistics(cfg, data);
stat_time.label = stat.label;

% just quick and dirty combination across planar gradiometers:
stat_time.accuracy = (stat_time.accuracy(1:149,:)+stat_time.accuracy(150:end,:))./2;
stat_time.label = label;

cfg.connectivity = C;
cfg.mvpa.timwin = 1; % otherwise the code currently crashes, but still it now does not give the desired result, to be fixed
stat_sens = ft_timelockstatistics(cfg, data);
stat_sens.label = stat.label;


cfg = [];
cfg.parameter = 'accuracy';
cfg.layout = 'CTF151_helmet.mat';
ft_topoplotER(cfg, stat_time,  stat);

**Comment by treder** [comment]
I've just posted a new pull request which resides in a different branch. With this in mind I close the current PR.
